src/Symfony/Component/Ldap/Adapter/ExtLdap/Query.php

```diff
@@ -76,39 +82,117 @@ public function execute()
+            $itemsLeft = $maxItems = $this->options['maxItems'];
+            $pageSize = $this->options['pageSize'];
+            if (0 !== $maxItems && $pageSize > $maxItems) {
+                $pageSize = 0;
+            } elseif (0 !== $maxItems) {
+                $pageSize = min($maxItems, $pageSize);
+            }
+            $pageControl = $this->options['scope'] != static::SCOPE_BASE && $pageSize > 0;
+            $cookie = '';
+            do {
+                if ($pageControl) {
+                    ldap_control_paged_result($con, $pageSize, true, $cookie);
+                }
+                $sizeLimit = $itemsLeft;
+                if ($pageSize > 0 && $sizeLimit >= $pageSize) {
+                    $sizeLimit = 0;
+                }
+                $search = @$func(
+                    $con,
+                    $this->dn,
+                    $this->query,
+                    $this->options['filter'],
+                    $this->options['attrsOnly'],
+                    $sizeLimit,
+                    $this->options['timeout'],
+                    $this->options['deref']
+                );
+
+                if (false === $search) {
+                    $ldapError = '';
+                    if ($errno = ldap_errno($con)) {
+                        $ldapError = sprintf(' LDAP error was [%d] %s', $errno, ldap_error($con));
+                    }
+                    if ($pageControl) {
+                        $this->resetPagination();
+                    }
+
+                    throw new LdapException(sprintf('Could not complete search with dn "%s", query "%s" and filters "%s".%s', $this->dn, $this->query, implode(',', $this->options['filter']), $ldapError));
+                }
+
+                $this->results[] = $search;
+                $itemsLeft -= min($itemsLeft, $pageSize);
+
+                if (0 !== $maxItems && 0 === $itemsLeft) {
+                    break;
+                }
+                if ($pageControl) {
+                    ldap_control_paged_result_response($con, $search, $cookie);
+                }
+            } while (null !== $cookie && '' !== $cookie);
+
+            if ($pageControl) {
+                $this->resetPagination();
             }
-
-            throw new LdapException(sprintf('Could not complete search with dn "%s", query "%s" and filters "%s".%s', $this->dn, $this->query, implode(',', $this->options['filter']), $ldapError));
         }
 
         return new Collection($this->connection, $this);
     }
 
     /**
-     * Returns a LDAP search resource.
+     * Returns a LDAP search resource. If this query resulted in multiple searches, only the first
+     * page will be returned.
      *
      * @return resource
      *
      * @internal
      */
-    public function getResource()
+    public function getResource($idx = 0)
+    {
+        if (null === $this->results || $idx >= \count($this->results)) {
+            return null;
+        } else {
```

Comment:
`else` can be remove as you done `return` in `if`

Reply:
Addressed in latest version; thanks!